src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,216 @@
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement updateCarStatement
+                        = connection.prepareStatement(query)) {
+            updateCarStatement.setString(1, car.getModel());
+            updateCarStatement.setLong(2, car.getManufacturer().getId());
+            updateCarStatement.setLong(3, car.getId());
+            updateCarStatement.executeUpdate();
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Couldn't update "
+                    + car + " in cars DB.", throwable);
+        }
+        deleteRelationsFromCarsDriversTable(car.getId());
+        insertDrivers(car);
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE cars SET is_deleted = TRUE WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement deleteCarStatement = connection.prepareStatement(query)) {
+            deleteCarStatement.setLong(1, id);
+            return deleteCarStatement.executeUpdate() > 0;
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Couldn't delete car with id " + id, throwable);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String query = "SELECT drivers.id, drivers.name AS driver_name, license_number, "
+                + "cars_drivers.car_id, cars.model, cars.manufacturer_id, "
+                + "manufacturers.name AS manufacturer, "
+                + "manufacturers.country AS manufacturer_country "
+                + "FROM drivers "
+                + "JOIN cars_drivers ON drivers.id = cars_drivers.driver_id "
+                + "JOIN cars ON cars_drivers.car_id = cars.id "
+                + "JOIN manufacturers ON cars.manufacturer_id = manufacturers.id "
+                + "WHERE cars_drivers.driver_id = ? AND drivers.is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getAllByDriverStatement = connection.prepareStatement(query)) {
+            getAllByDriverStatement.setLong(1, driverId);
+            ResultSet resultSet = getAllByDriverStatement.executeQuery();
+            List<Car> cars = new ArrayList<>();
+            while (resultSet.next()) {
+                Car car = parseCarWithManufacturerFromResultSet(resultSet);
+                List<Driver> driverList = new ArrayList<>();
+                driverList.add(parseDriversFromResultSet(resultSet));
+                car.setDrivers(driverList);
+                cars.add(car);
+            }
+            return cars;
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Couldn't get cars by driver id "
+                    + driverId, throwable);
+        }
+    }
+
+    private void insertDrivers(Car car) {
```

Comment:
according to your naming style, better name it createReleationsCarsDriversTable or smth